.github/workflows/main.yml

```
@@ -72,6 +72,68 @@ jobs:
       - name: Run Tests
         run: yarn test
 
+      - name: Consolidate Coverage Reports
+        run: |
+          echo "🔍 Consolidating coverage reports for SonarQube..."
+          echo "================================================="
+
+          # Create consolidated coverage directory
+          mkdir -p sonar-coverage
+
+          # Find all lcov.info files
+          echo "1. Finding coverage reports..."
+          coverage_files=$(find packages -name "lcov.info" -type f)
+          echo "Found coverage files:"
+          echo "$coverage_files"
+
+          if [ -z "$coverage_files" ]; then
+              echo "❌ No coverage files found!"
+              exit 1
+          fi
+
+          # Consolidate all lcov.info files into one with corrected paths
+          echo ""
+          echo "2. Consolidating coverage reports with absolute paths..."
+          consolidated_file="sonar-coverage/lcov.info"
+
+          # Process each file and fix paths
+          for file in $coverage_files; do
+              if [ -f "$file" ] && [ -s "$file" ]; then
+                  echo "Processing $file"
+                  # Get the package directory (e.g., packages/react-packages/tabs from packages/react-packages/tabs/jest-coverage/lcov.info)
+                  package_dir=$(dirname $(dirname $file))
+                  echo "  Package directory: $package_dir"
+                  # Process the lcov file and update paths to be relative to project root
+                  sed "s|^SF:src/|SF:$package_dir/src/|g; s|^SF:dist/|SF:$package_dir/dist/|g; s|^SF:lib/|SF:$package_dir/lib/|g; s|^SF:build/|SF:$package_dir/build/|g" "$file" >> "$consolidated_file"
+              fi
+          done
+
+          echo "✅ Consolidated coverage report created: $consolidated_file"
+
+          # Show summary
+          echo ""
+          echo "3. Coverage report summary:"
+          if [ -f "$consolidated_file" ]; then
+              lines=$(wc -l < "$consolidated_file")
+              echo "Total lines in consolidated report: $lines"
+              
+              # Show first few lines
+              echo "First 10 lines:"
+              head -10 "$consolidated_file"
+          else
+              echo "❌ Consolidated file not created!"
```

```suggestion
              echo "Consolidated file not created!"
```
I would removed all this icons on the logs

---

done @joelpereira15 will avoid sensorial overload from now on 😝